### PR TITLE
2386 replace twitter image

### PR DESCRIPTION
**Pull request by AndrewBarber**
#### What?

Updated Twitter/X imagery used within the Social Share and Social Links components, to be inline with new [Twitter/X brand guidelines](https://about.twitter.com/en/who-we-are/brand-toolkit).

#### Requirements

- [X] CHANGELOG.md entry added (required for code changes only)

#### Tickets / Documentation

- #2386 

#### Screenshots (if appropriate)

Screenshot 2023-08-12 at 13 45 32
Screenshot 2023-08-12 at 13 45 48

### CODE REVIEW

**Review by BC-krasnoshapka — APPROVED** [review]
LGTM!

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,7 +5,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
 ## Draft
-
+- Replace Twitter logo with X logo within social sharing and social link components [#2386]
```
Comment (bc-yevhenii-buliuk) [inline CHANGELOG.md:8]: @AndrewBarber Could you add a link after PR's number please
`(https://github.com/bigcommerce/cornerstone/pull/2387)`
and fix please number of PR
`2386 -> 2387`

Comment (AndrewBarber) [reply CHANGELOG.md:8]: Hey, sorry I don't understand what you're asking me here.

File: templates/components/common/social-links.html [hunk 2]
```diff
@@ -1,19 +1,35 @@
...
-                    </svg>
-                </a>
-            </li>
+            {{#if display_name '===' 'twitter'}}
+                <li class="socialLinks-item">
+                    <a class="icon icon--x"
+                    href="{{url}}"
+                    target="_blank"
+                    rel="noopener"
+                    title="{{{capitalize display_name}}}"
```
Comment (bc-yevhenii-buliuk) [inline templates/components/common/social-links.html:10]: @AndrewBarber in this case should we change `title="{{{capitalize display_name}}}"` to just `title="X"`?

Comment (AndrewBarber) [reply templates/components/common/social-links.html]: Agreed, will set static.

**Pull request merged**